lib/protocol/rest_json.js

```diff
@@ -4,6 +4,16 @@ var Json = require('./json');
 var JsonBuilder = require('../json/builder');
 var JsonParser = require('../json/parser');
 
+function unsetContentLength(req) {
+  var payloadMember = util.getRequestPayloadShape(req);
+  if (
+    payloadMember === undefined &&
+    ['GET', 'HEAD', 'DELETE'].indexOf(req.httpRequest.method) >= 0
```

Comment:
any need to add HEAD/OPTIONS?

Reply:
No as per internal RFC. On line 53, we avoid sending payload for GET, HEAD and DELETE

Reply:
Created a reusable array in commit [`2329b60` (#4219)](https://github.com/aws/aws-sdk-js/pull/4219/commits/2329b60df0850adcbac009375a23706b9a48fed4) to reduce duplicate code.